### PR TITLE
Enable using ImageSummary in SpmdEvaler.

### PR DESCRIPTION
**Pull request by apghml**

### CODE REVIEW

File: axlearn/common/module.py [hunk 1]
```diff
@@ -242,6 +242,13 @@ def add_summary(
             name: The name of the item to add.
             value: The value to add.
         """
+
```
Comment (markblee) [inline axlearn/common/module.py:245]: (Not from this PR, but just noticed that `add_summary` typing seems out of date.)

Comment (apghml) [reply axlearn/common/module.py:245]: Done.

File: axlearn/common/module_test.py [hunk 2]
```diff
@@ -738,6 +739,20 @@ def test_drop_output(self):
...
+        class MySummary(summary.Summary):
+            val: str
+
+            def validate(self):
+                if self.val == "s":
+                    raise ValueError("not allowed")
+
+        with self._dummy_context() as ctx:
+            with self.assertRaises(ValueError):
+                ctx.add_summary("summary", MySummary("s"))
```
[the remark this answers is not in the document]
Comment (apghml) [reply axlearn/common/module_test.py]: Done.

File: axlearn/common/summary_test.py [hunk 3]
```diff
@@ -56,6 +64,103 @@ def test_add_summary_image(self):
...
+                Expected(
+                    path=os.path.join(cfg.dir, "summaries", "train_train"), count=8, key="model/img"
+                ),
+            ]
+
+            for info in expected:
+                ea = event_accumulator.EventAccumulator(info.path)
+                ea.Reload()
+
+                print(ea.tensors.Keys())
```
[the remark this answers is not in the document]
Comment (apghml) [reply axlearn/common/summary_test.py]: Done, thanks.

File: axlearn/common/summary_test.py [hunk 4]
```diff
@@ -56,6 +64,103 @@ def test_add_summary_image(self):
...
+                                tf.image.decode_image(im)
+                                for im in tf.make_ndarray(event.tensor_proto)[2:]
+                            ]
+                        )
+                        for event in ea.Tensors(info.key)
+                    ]
+                )
+                self.assertEqual(logged_evaler_img.shape, info.shape())
+                # TB uses lossy compression.
+                chex.assert_trees_all_close(logged_evaler_img / 255, info.img(), rtol=0.01)
```
[the remark this answers is not in the document]
Comment (apghml) [reply axlearn/common/summary_test.py]: Done.

**Comment by apghml** [comment]
Soft ping @markblee.

**Review by ruomingp — APPROVED** [review]

**Pull request merged**